datachannel_test.go

```diff
@@ -56,6 +56,41 @@ func closeReliabilityParamTest(t *testing.T, pc1, pc2 *PeerConnection, done chan
 	closePair(t, pc1, pc2, done)
 }
 
+func TestDataChannel_Open(t *testing.T) {
+	t.Run("handler should be called once", func(t *testing.T) {
+		report := test.CheckRoutines(t)
+		defer report()
+
+		offerPC, answerPC, err := newPair()
+		if err != nil {
+			t.Fatalf("Failed to create a PC pair for testing")
+		}
+
+		openCalls := make(chan bool, 2)
+
+		answerPC.OnDataChannel(func(d *DataChannel) {
+			if d.Label() != expectedLabel {
+				return
+			}
+			d.OnOpen(func() {
+				openCalls <- true
+			})
+		})
+
+		_, err = offerPC.CreateDataChannel(expectedLabel, nil)
+		assert.NoError(t, err)
+
+		assert.NoError(t, signalPair(offerPC, answerPC))
+
+		// wait for open handlers to be executed
+		time.Sleep(1 * time.Second)
```

Comment:
Tests which rely sleeping for a certain amount of time like this can be finicky. For example, on slower machines, it might take longer than 1 second for the open handlers to be executed. Is there anyway we could avoid this?

Reply:
yes, I don't like it either, but couldn't find a way to avoid it, I'm open to suggestions

Reply:
This is a tough one, I have no idea how to properly test. You are almost trying to assert `This thing shouldn't happen, but I don't know when it will happen`

Oh, what do you think of making a `maybeFireOnOpenHandler` (or a better name) and call it 3 times (asserting that your callback only gets fired once) 

Someone can always circumvent it and call the handler manually, but maybe just adding a note is good enough.